alphabase/constants/atom.py

```diff
@@ -200,3 +204,195 @@ def calc_mass_from_formula(formula: str):
         mass of the formula
     """
     return np.sum([CHEM_MONO_MASS[elem] * n for elem, n in parse_formula(formula)])
+
+
+class ChemicalCompositonFormula:
+    """
+    Initialize the ChemicalCompositonFormula with a given formula.
+
+    Parameters
+    ----------
+    formula : str
+        The chemical formula as a string.
+
+    Returns
+    -------
+    None
+    """
+
+    def __init__(self, formula=None):
+        self.elements = (
+            defaultdict(int) if formula is None else self._parse_formula(formula)
```

Comment:
I am still not sure if having a `defaultdict` as instance variable type is a good idea (I guess it was mine) .. I think the user of the class will not know what it is, and on wrong use, there will be no `KeyErrors` but much less visible zeros..
what do you think?

As a quick remedy, one could convert the `defaultdict` to `dict` when returning from `parse_formula` & `parse_rdkit_formula`

Reply:
added a method to make sure elements of the defaultdict are valid atom names